File: frontend/lib/features/training/domain/models/exercise.dart
class Exercise {
  final String id;
  final String name;
  final String? muscleGroup;
  final String? equipment;
  final String? description;

  Exercise({
    required this.id,
    required this.name,
    this.muscleGroup,
    this.equipment,
    this.description,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['id'],
      name: json['name'],
      muscleGroup: json['muscleGroup'],
      equipment: json['equipment'],
      description: json['description'],
    );
  }
}

File: backend/src/main/java/com/ironpath/backend/training/domain/model/Exercise.java
package com.ironpath.backend.training.domain.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "exercises")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Exercise {

    @Id
    private String id;

    @Column(nullable = false)
    private String name;

    @Column(name = "muscle_group")
    private String muscleGroup;

    private String equipment;

    @Column(columnDefinition = "TEXT")
    private String description;
}